src/testrun.rs

```diff
@@ -186,7 +178,9 @@ impl Testrun {
                 && self.outcome == other.outcome
                 && self.duration == other.duration
                 && self.testsuite == other.testsuite
-                && self.failure_message == other.failure_message),
+                && self.failure_message == other.failure_message
```

Comment:
I’m a bit unsure why you need to implement `__richcmp__` yourself here. But on the Rust side, you can derive `PartialEq`.

Although you are missing `build_url` in this comparison. And floating point comparison of `duration` is also most likely something you would want to avoid.

Reply:
i don't remember exactly why, but i think Testrun already derives PartialEq and if i remove it, it breaks the test, i think i can take another look at it after this merges